src/components/AuthContainer.js

```diff
@@ -9,15 +9,15 @@ import TermsOfService from 'src/pages/TermsOfService'
 
 
 export default Utils.connectAtom(authStore, 'authState')(
-  ({ children, authState: { isSignedIn, registrationStatus, acceptedTos } }) => {
+  ({ children, isPublic, authState: { isSignedIn, registrationStatus, acceptedTos } }) => {
     return Utils.cond(
```

Comment:
The only additional thought I had for improving this is that maybe `Utils.cond` is no longer the best way to express this logic. There's an implied if/else structure here that might be more clear if expanded out.

Reply:
Do you mean an if/else like this?
```
return !isPublic && (isSignedIn === undefined || registrationStatus === undefined || acceptedTos === undefined) ?
      centeredSpinner
      : (isSignedIn === false && !isPublic ?
        h(SignIn)
        : (registrationStatus === 'unregistered' ?
          h(Register)
          : (registrationStatus === 'disabled' ?
            () => h(Disabled)
            : (registrationStatus === 'unlisted' ?
              () => h(Unlisted)
              : (acceptedTos === false ?
                () => h(TermsOfService)
                : children
              )))))
```
I couldn't get the logic to quite work out, I think because I didn't switch from `Utils.cond` correctly. In the interest of time I'm going to merge my fix to see if that helps with login errors.